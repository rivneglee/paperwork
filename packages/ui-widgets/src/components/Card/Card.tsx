import React from 'react';

export default () => (
  <div className="card">
    Card
  </div>
);
