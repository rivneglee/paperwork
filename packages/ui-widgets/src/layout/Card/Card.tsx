import React, { ReactComponentElement, FunctionComponent } from 'react';

interface Props {
  children: ReactComponentElement<any> | string;
  header?: ReactComponentElement<any> | string;
  footer?: ReactComponentElement<any> | string;
}

const Card: FunctionComponent<Props> = ({
 children, header, footer,
}) => (
  <div className="pw-card">
    {
      header && (
        <div className="pw-card__header">
          {header}
        </div>
      )
    }
    <div className="pw-card__body">
      {children}
    </div>
    {
      header && (
        <div className="pw-card__footer">
          {footer}
        </div>
      )
    }
  </div>
);

export default Card;
