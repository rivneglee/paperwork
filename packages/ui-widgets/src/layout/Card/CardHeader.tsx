import React, { FunctionComponent, ReactElement } from 'react';

interface Props {
  icon?: ReactElement;
  primary?: ReactElement | string;
  secondary?: ReactElement | string;
  className?: string;
}

export type CardHeaderComponent = FunctionComponent<Props>;

const CardHeader: CardHeaderComponent = ({
 children, primary, secondary, icon, className = '',
}) => (
  <div className={`pw-card-header ${className}`}>
    <span className="pw-card-header__icon">{icon}</span>
    <h3 className="pw-card-header__primary">
      {primary}
    </h3>
    <div className="pw-card-header__secondary">
      {secondary}
    </div>
  </div>
);

export default CardHeader;
