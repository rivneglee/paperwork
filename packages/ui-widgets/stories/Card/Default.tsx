import React from 'react';

import { Card } from '../../src';

export default () => <Card />;
