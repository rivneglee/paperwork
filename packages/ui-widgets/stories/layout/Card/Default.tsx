import React from 'react';

import { Card } from '../../../src';

export default () => (
  <div style={{ background: '#ebeef1', padding: '2.4rem' }}>
    <Card>
      <div>This is a card</div>
    </Card>
  </div>
);
