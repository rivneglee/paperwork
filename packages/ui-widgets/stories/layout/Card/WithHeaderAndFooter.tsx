import React from 'react';

import { Card } from '../../../src';

export default () => (
  <div style={{ background: '#ebeef1', padding: '2.4rem' }}>
    <Card header="Header" footer="Footer">
      <div>This is a card</div>
    </Card>
  </div>
);
