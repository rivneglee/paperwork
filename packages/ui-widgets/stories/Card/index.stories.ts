import { storiesOf } from '@storybook/react';

import Default from './Default';

storiesOf('Card', module)
  .add('default', Default);
