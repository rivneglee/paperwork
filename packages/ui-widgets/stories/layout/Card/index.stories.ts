import { storiesOf } from '@storybook/react';

import Default from './Default';
import WithHeaderAndFooter from './WithHeaderAndFooter';

storiesOf('Card', module)
  .add('default', Default)
  .add('withHeaderAndFooter', WithHeaderAndFooter);
