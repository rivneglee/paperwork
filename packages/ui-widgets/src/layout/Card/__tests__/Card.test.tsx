import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import Card from '../Card';

describe('Card', () => {
  const Header = () => <>'Header'</>;
  const Footer = () => <>'Footer'</>;
  const Content = () => <>'Content'</>;

  it('should render with header and footer', () => {
    const wrapper: ShallowWrapper = shallow(
      <Card header={<Header />} footer={<Footer />}>
        <Content />
      </Card>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without header and footer', () => {
    const wrapper: ShallowWrapper = shallow(
      <Card>
        <Content />
      </Card>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
