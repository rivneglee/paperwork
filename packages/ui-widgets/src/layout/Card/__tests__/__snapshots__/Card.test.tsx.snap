// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render with header and footer 1`] = `
<div
  className="pw-card"
>
  <div
    className="pw-card__header"
  >
    <Header />
  </div>
  <div
    className="pw-card__body"
  >
    <Content />
  </div>
  <div
    className="pw-card__footer"
  >
    <Footer />
  </div>
</div>
`;

exports[`Card should render without header and footer 1`] = `
<div
  className="pw-card"
>
  <div
    className="pw-card__body"
  >
    <Content />
  </div>
</div>
`;
